<p><img src="../IMGs/logo-fing.png" alt="LogoUSACH" width="40%" align="right" hspace="10px" vspace="0px"></p>

# INTRODUCCIÓN A PYTHON: TIPOS DE DATOS, VARIABLES, OPERADORES Y EXPRESIONES




>#### 💡 **Recuerda:** para guardar tus cambios, ve a **Archivo → Guardar una copia en Drive**. Así tendrás tu propia copia editable y no perderás lo que hagas.


## **Expresiones Matemáticas**

Antes de comenzar a trabajar en Python a nivel general, es necesario aprender a realizar operaciones aritméticas básicas.

### Expresiones Binarias

La notación para realizar operaciones aritméticas básicas **binarias** es la siguiente:

**Número -->  Operador -->  Número**

In [1]:
# Ejemplo 1: Suma
5 + 4

9

In [2]:
# Ejemplo 2: Suma
8.0 + 3

11.0

In [3]:
# Ejemplo 3: Resta
3 - 7

-4

In [4]:
# Ejemplo 4: Resta
10 - 5.4

4.6

Los ejemplos anteriores se conocen como **notación infija**, y es la que
se debe utilizar cada vez que se escriban operaciones matemáticas.

Hay que considerar que para realizar una `multiplicación` se debe utilizar el operador * (**asterisco**) y para realizar una `división` se debe utilizar el operador / (**slash**)

In [5]:
# Ejemplo 5: Multiplicación
2*4

8

In [6]:
# Ejemplo 6: Multiplicación
2.5*5

12.5

In [7]:
# Ejemplo 7: División
10/5

2.0

In [8]:
# Ejemplo 8: División
12/8

1.5

Además de las 4 operaciones básicas, Python provee operaciones que pueden ser de utilidad:

*   **Potencia** (`**`)
*   **Resto o módulo** (**`%`**)
*   **División parte entera** (**`//`**)


In [9]:
# Ejemplo 9: Potencia
2**4

16

In [10]:
# Ejemplo 10: Potencia
3**2

9

In [11]:
# Ejemplo 11: Resto o módulo
10%3

1

In [12]:
# Ejemplo 12: Resto o módulo
24%6

0

In [13]:
# Ejemplo 13: División parte entera
10//3

3

In [14]:
# Ejemplo 14: División parte entera
4//6

0

### Expresiones Unarias



Por otra parte, también se pueden encontrar **operadores unarios**:
*  **Identidad** (**`+`**)
*  **Negación** (**`-`**)

In [15]:
# Ejemplo 15: Identidad
+4

4

In [16]:
# Ejemplo 16: Identidad
+(-23)

-23

In [17]:
# Ejemplo 17: Negación
-6

-6

In [18]:
# Ejemplo 18: Negación
-(-5)

5

### Precedencia de Operadores


La `precedencia de operadores` son reglas sencillas que permiten a Python "ordenar" la evaluación de expresiones. Se reconocen 3 tipos de operadores:
1. **Operadores aritméticos**: corresponden a los operadores que se han visto hasta el momento y sirven para realizar cálculos matemáticos sencillos.
2. **Operadores de comparación**: sirven para hacer evaluaciones que siempre resultan en valores de verdad. Estos serán revisados más adelante, cuando se quiera condicionar los programas.
3. **Operadores lógicos**: sirven para representar las combinaciones que pueden hacerse sobre valores de verdad y que también se verán más adelante.

Para Python, siempre se verán primero los operadores `aritméticos`, luego los de `comparación` y finalmente los `lógicos`. Además, cada categoría tiene su propio orden de evaluación preestablecido.

Python sigue las reglas de precedencia de signos, las cuales fueron vistas en la asignatura de matemáticas.

Antes de continuar, resuelve en tu cuaderno la siguiente expresión matemática:

5 + 5 ** 3 / 5 * 4 - 10 * 3

In [19]:
# Ejemplo 19: Compara el resultado obtenido con lo que arroja Python
5+5**3/5*4-10*3

75.0

Al resolver una expresión a mano y luego compararla con el resultado que entrega el intérprete de Python, ambos resultados pueden diferir. Esto ocurre por dos motivos:


*   La precedencia de operadores
*   El tipo de dato



En los operadores aritméticos, el orden de prioridad, desde el operador más importante al de menor importancia, se presenta en la siguiente tabla:




|Operación     |Operador      |Aridad        |Asociatividad |Precedencia|
|:------------:|:------------:|:------------:|:------------:|:---------:|
|Exponenciación| `**`         |Binaria       |Derecha       |1          |
|Identidad     | `+`          |Unaria        |-             |2          |
|Negación      | `-`          |Unaria        |-             |2          |
|Multiplicación| `*`         |Binaria       |Izquierda     |3          |
|División      | `/`         |Binaria       |Izquierda     |3          |
|Módulo o resto       | `%`         |Binaria       |Izquierda     |3          |
|División parte entera| `//`         |Binaria       |Izquierda |3         |
|Suma| `+`         |Binaria       |Izquierda       |4          |
|Resta| `-`         |Binaria       |Izquierda      |4          |

Las reglas de precedencia pueden ser modificadas utilizando ( ) **paréntesis**.

A modo de ejemplo, si se quisiera realizar el ejercicio anterior

5 + 5 ** 3 / 5 * 4 - 10 * 3

pero darle prioridad a la operación 5 + 5, se debe utilizar **paréntesis**

(5 + 5) ** 3 / 5 * 4 - 10 * 3



In [20]:
# Ejemplo 20: Vuelve a desarrollar el ejercicio sin paréntesis
5 + 5 ** 3 / 5 * 4 - 10 * 3

75.0

In [21]:
# Ejemplo 21: Desarrolla el ejercicio con paréntesis
(5 + 5) ** 3 / 5 * 4 - 10 * 3

770.0

Cuando se quiere modificar las reglas de precedencia pueden surgir los siguientes problemas:


*   Los paréntesis que se abren deben cerrarse, vale decir, los paréntesis deben estar **balanceados**.

*   Solo se deben utilizar paréntesis redondos ( ). Los `corchetes` [ ] y las `llaves` { } se usan para definir otros tipos de datos: las **listas** (que se verán más adelante en el curso) y los **diccionarios** (que no forman parte de este curso).

Veamos el siguiente ejemplo, que **genera un error de forma intencional**, ya que sus paréntesis no están balanceados:

In [22]:
# Ejemplo 22
(2+((3**2)/5*3)+3*2-(3*4)+5)+5)

SyntaxError: unmatched ')' (736502302.py, line 2)

## **Tipos de datos**


En todos los lenguajes de programación existen distintos tipos de datos para facilitar el trabajo a realizar. Los tipos de datos son un conjunto determinado de valores con propiedades y características determinadas.

Los 3 tipos principales de datos son:

*   Tipo numérico
*   Tipo texto
*   Tipo lógico

Los tipos de datos texto y lógicos serán vistos con mayor detalle a medida que avance el curso, por el momento solo se dará una pequeña definición de ellos para comprender su importancia.


### Tipo numérico

En Python existen tres clases de números:

*   **Números enteros** (**`int`**): son una representación de los números enteros, tanto positivos como negativos.

*   **Números flotantes** (**`float`**): son una **aproximación** de los números reales, vale decir, son números que tienen una parte entera y una parte decimal. Se debe tener en cuenta que para separar la parte entera de la decimal se utiliza el punto (.), esto se debe a que Python fue escrito por angloparlantes. A los números flotantes se les denomina números de punto flotante.

*   **Números complejos** (**`complex`**): son una representación de conjunto matemático correspondiente a los números complejos. Lo anterior significa que el número está conformado por una parte real y otra imaginaria. Esta última es acompañada con la letra **`j`**

Cabe señalar que los tipos de datos numéricos son solo aproximaciones de los conjuntos matemáticos que representan debido a que la cardinalidad de estos es infinita y los computadores tienen una cantidad de memoria finita

In [23]:
# Ejemplo 23: Operación Números Complejos
(2+5j) + (4-1j)


(6+4j)

In [24]:
# Ejemplo 24: Operación Números Complejos
15 - (4-1j)

(11+1j)

In [25]:
# Ejemplo 25: Operación Números Enteros y Flotantes
2 + 4 - 8 + 15.0


13.0

En el caso del último ejemplo, al desarrollar la expresión matemática entrega como resultado un número flotante. Esto se debe a que cuando se tengan operaciones que combinen distintos tipos de datos numéricos, Python **siempre mantendrá el resultado en el conjunto más general de los conjuntos numéricos**.

Por ello, dado que en el ejercicio anterior se tienen datos de tipo `entero` y `flotante`, el resultado será `flotante`.

#### Los flotantes no son los números reales

Es tentador pensar que el tipo `float` "es" el conjunto de los números reales, pero **no lo es**. Un computador guarda cada número usando una cantidad **finita** de memoria, mientras que representar un número real cualquiera requiere, en general, infinitos dígitos. Lo que Python almacena en un `float` es entonces una **aproximación binaria** del número que nosotros escribimos en decimal.

Esto tiene dos consecuencias que van a aparecer durante todo el curso:

1.  **Hay números decimales que no se pueden representar de forma exacta.** Así como en base 10 el número 1/3 necesita infinitos decimales (0.3333...), en base 2 le ocurre lo mismo a números tan simples como 0.1.
2.  **El error de representación se acumula al operar.**

Observa el resultado de la siguiente suma:


In [26]:
# Ejemplo 26: Un número decimal aparentemente simple no se guarda de forma exacta
0.1 + 0.2


0.30000000000000004

El resultado no es `0.3`, sino `0.30000000000000004`. Ese resto diminuto es la diferencia entre el número que escribimos y el número que el computador efectivamente guardó.

Si la operación se repite, el error se arrastra y se hace más visible:


In [27]:
# Ejemplo 27: El error de representación se acumula
0.1 + 0.1 + 0.1 + 0.1 + 0.1 + 0.1 + 0.1 + 0.1 + 0.1 + 0.1


0.9999999999999999

A esto se suma un segundo límite: los flotantes tienen una **precisión máxima**. Pasado cierto tamaño, el número guardado es tan grande que sumarle 1 ya no alcanza a cambiar el valor almacenado:


In [28]:
# Ejemplo 28: A partir de cierto tamaño, sumar 1 deja de tener efecto sobre un float
9007199254740992.0 + 1


9007199254740992.0

Nada de esto es un error de Python: es la consecuencia inevitable de representar números infinitos en una memoria finita, y ocurre prácticamente igual en todos los lenguajes de programación.

Las consecuencias prácticas para este curso son tres:

*   Los `float` sirven para **medir** (distancias, promedios, tiempos). Para **contar** se usa `int`, que sí es exacto y no tiene un límite de precisión de este tipo.
*   Nunca hay que suponer que dos cálculos con flotantes que "deberían" dar lo mismo entregarán exactamente el mismo valor. En el apunte de decisiones se verá cómo comparar flotantes correctamente.
*   Si un resultado aparece con una cola de decimales extraña (como `0.30000000000000004`), lo más probable es que no sea un error del programa, sino el comportamiento normal del tipo `float`.


### Tipo texto

Son conocidos como **strings** (**`str`**) y se utilizan para almacenar secuencias de caracteres, usadas para la generación de mensajes y archivos con información

Los strings se identifican con la utilización de comillas simples (**'**) o comillas dobles (**"**), las cuales delimitan el comienzo y fin de la secuencia de caracteres.

Cabe señalar que, si el string se inicia con comillas simples, debe terminar con comillas simples. Si se inicia con comillas dobles, debe terminar con comillas dobles.

In [29]:
# Ejemplo 29: Ejemplo string
"este es un string"

'este es un string'

In [30]:
# Ejemplo 30: Ejemplo string
'este es un string'

'este es un string'

In [31]:
# Ejemplo 31: Ejemplo string: esta celda genera un error de forma intencional
'este es un mal ejemplo de string"

SyntaxError: unterminated string literal (detected at line 2) (4106335299.py, line 2)

### Tipo lógico


También denominado booleano, es un tipo de dato que permite expresar 2 valores: Verdadero (***True***) y  Falso (***False***)

De manera interna, el valor **False** representa un **0** y el Valor **True** equivale a un **1**

Cabe destacar que ambos valores se deben escribir capitalizados, vale decir, su primera letra debe ser mayúscula para que Python los reconozca como valores lógicos.

Los dos primeros ejemplos muestran la escritura correcta. Los dos siguientes **generan un error de forma intencional**, al escribir los valores en minúscula: Python no los reconoce como valores lógicos, sino como nombres de variables que nunca fueron definidas.

In [32]:
# Ejemplo 32: Ejemplo booleano
True

True

In [33]:
# Ejemplo 33: Ejemplo booleano
False

False

In [34]:
# Ejemplo 34: Ejemplo booleano
true

NameError: name 'true' is not defined

In [35]:
# Ejemplo 35: Ejemplo booleano
false

NameError: name 'false' is not defined

## **Conversión de Tipos**

Como se vio anteriormente, Python reconoce el tipo de dato con el cual se quiere trabajar y en el caso del tipo de dato numérico, se pueden mezclar para generar nuevas expresiones aritméticas, permitiendo que Python se quede con el tipo de dato más general.

No obstante, existe la posibilidad de poder indicar a Python qué tipo de dato se quiere obtener de manera directa. Para realizar dicho cometido, se utilizarán funciones nativas (más adelante se explicarán en detalle).

Las funciones de cambio de tipo (*typecast*) a utilizar son las siguientes:


*   Conversión a número entero      --->  `int( )`

*   Conversión a número flotante    --->  `float( )`

*   Conversión a número complejo    --->  `complex( )`

*   Conversión a texto    --->  `str( )`

*   Conversión a booleano    --->  `bool( )`



**No todas las conversiones son posibles.** Algunas de las celdas que siguen generan errores de forma intencional: la idea es que reconozcan el mensaje que entrega Python y entiendan por qué esa conversión en particular no tiene sentido.


### Conversión a número entero

In [36]:
# Ejemplo 36: Conversión número flotante a entero
int(3.6)

3

In [37]:
# Ejemplo 37: Conversión número complejo a entero
int(5-3j)

TypeError: int() argument must be a string, a bytes-like object or a real number, not 'complex'

In [38]:
# Ejemplo 38: Conversión texto a número entero
int("5")

5

In [39]:
# Ejemplo 39: Conversión booleano a entero
int(False)

0

La conversión `int(5-3j)` **genera un error de forma intencional**. Python responde `TypeError: can't convert complex to int`, porque un número complejo está formado por dos componentes (una real y una imaginaria) y no hay una forma única de transformarlo en un único número entero.

En cambio, `int(3.6)` sí funciona, y conviene notar que el resultado es `3` y no `4`: la conversión a entero **trunca** el número, es decir, elimina la parte decimal en lugar de aproximar al entero más cercano.


### Conversión a número flotante

In [40]:
# Ejemplo 40: Conversión número entero a flotante
float(5)

5.0

In [41]:
# Ejemplo 41: Conversión número complejo a flotante
float(5j)

TypeError: float() argument must be a string or a real number, not 'complex'

In [42]:
# Ejemplo 42: Conversión texto a número flotante
float("4.2")

4.2

In [43]:
# Ejemplo 43: Conversión booleano a número flotante
float(True)

1.0

La conversión `float(5j)` **genera un error de forma intencional**, por la misma razón que el caso anterior: un número complejo tiene dos componentes y no existe una forma única de reducirlo a un solo número flotante.


### Conversión a número complejo

In [44]:
# Ejemplo 44: Conversión número entero a complejo
complex(5)

(5+0j)

In [45]:
# Ejemplo 45: Conversión número flotante a complejo
complex(2.8)

(2.8+0j)

In [46]:
# Ejemplo 46: Conversión texto a número complejo
complex("1.5-j")

(1.5-1j)

In [47]:
# Ejemplo 47: Conversión booleano a número complejo
complex(True,False)

(1+0j)

Estas dos últimas conversiones merecen atención:

*   `complex("1.5-j")` **sí funciona** y entrega `(1.5-1j)`. Python acepta construir un complejo desde un string, siempre que este tenga el formato de un número complejo y **no contenga espacios**. La `j` sin coeficiente se interpreta como `1j`.
*   `complex(True, False)` muestra que esta función también puede recibir **dos argumentos**: el primero pasa a ser la parte real y el segundo la parte imaginaria. Como `True` equivale a 1 y `False` a 0, el resultado es `(1+0j)`.


### Conversión a texto

In [48]:
# Ejemplo 48: Conversión número entero a texto
str(4)

'4'

In [49]:
# Ejemplo 49: Conversión número flotante a texto
str(2.8)

'2.8'

In [50]:
# Ejemplo 50: Conversión número complejo a texto
str(8.1+20j)

'(8.1+20j)'

In [51]:
# Ejemplo 51: Conversión booleano a texto
str(True)

'True'

### Conversión a booleano

In [52]:
# Ejemplo 52: Conversión número entero a booleano
bool(34)

True

In [53]:
# Ejemplo 53: Conversión número flotante a booleano
bool(9.56)

True

In [54]:
# Ejemplo 54: Conversión número complejo a booleano
bool(0+0j)

False

In [55]:
# Ejemplo 55: Conversión número complejo a booleano
bool(1+1j)

True

In [56]:
# Ejemplo 56: Conversión texto a booleano
bool("False")

True

In [57]:
# Ejemplo 57: Conversión texto a booleano
bool("True")

True

Los dos últimos ejemplos suelen sorprender: **`bool("False")` entrega `True`**.

Esto ocurre porque la conversión a booleano no interpreta el *contenido* del texto, sino que se pregunta si el dato está **vacío**:

*   Cualquier string con al menos un caracter se convierte en `True`, aunque ese caracter sea un `0` o la palabra `False`.
*   Solo el string vacío (`""`) se convierte en `False`.

La misma regla aplica a los números: el `0` (en cualquiera de sus tipos) se convierte en `False` y cualquier otro valor en `True`.


In [58]:
# Ejemplo 58: Conversión de un string vacío a booleano
bool("")


False

## **Operador asignación**

Además de los operadores mencionados anteriormente, Python posee un operador especial denominado asignación (**=**). Dicho operador permite **almacenar** un valor o resultado de una expresión con un **nombre determinado por el/la programador/a** para posteriormente usarlo.

La estructura del operador asignación es la siguiente:
```
                                           <identificador> = <expresión>
```
Se debe tener en consideración que el símbolo de asignación (**=**) NO es una igualdad o equivalencia, sino que permite guardar un valor/expresión determinada.

Conociendo el concepto del operador asignación, se pueden definir las **variables** y **constantes** para almacenar valores.


> ### ⚠️ Mala práctica en este curso: las asignaciones múltiples
>
> Python permite asignar varios valores en una sola línea, separando tanto los identificadores como las expresiones con comas:
>
> ```python
> lado_a, lado_b = 3, 4
> ```
>
> Es una sintaxis válida y bastante usada por programadores con experiencia, pero **en este curso no se va a utilizar**. La razón es que oculta la lógica del programa: obliga a quien lee el código a emparejar mentalmente cada identificador con la expresión que le corresponde, y ese emparejamiento se vuelve difícil de seguir apenas la línea crece o las expresiones dejan de ser valores simples.
>
> La forma que se usará en el curso es **una asignación por línea**:
>
> ```python
> lado_a = 3
> lado_b = 4
> ```


### **Variables y Expresiones**

#### Reglas para identificadores

Las variables y constantes pueden tener cualquier nombre, siempre y cuando se cumplan algunas sencillas reglas:


1.   Un identificador puede estar conformado por:

   * **Letras**: minúsculas y/o mayúsculas
   * **Dígitos**: 0 - 9
   * Caracter de subrayado o **guión bajo**

2.   El **primer caracter** no puede ser un dígito

3.   No puede coincidir con una **palabra reservada** del lenguaje Python, las cuales son: **False**, **None**, **True**, **and**, **as**, **assert**, **async**, **await**, **break**, **class**, **continue**, **def**, **del**, **elif**, **else**, **except**, **finally**, **for**, **from**, **global**, **if**, **import**, **in**, **is**, **lambda**, **nonlocal**, **not**, **or**, **pass**, **raise**, **return**, **try**, **while**, **with**, **yield**. El editor las destaca automáticamente con un color distinto al del resto del código.

   Distinto es el caso de los nombres de las **funciones nativas** (como `print`, `len` o `str`): no son palabras reservadas, por lo que Python permite usarlas como identificador, pero al hacerlo la función deja de estar disponible en el resto del programa. Por eso tampoco se deben usar como nombre de una variable.

4.  Debe ser representativo, ya que hace más entendible el programa. Por ejemplo, "radio", "sumando", "nombre", "edad".

5. Para el caso de las **constantes**, sus nombres se escriben siempre con mayúscula y separando las palabras con el caracter de subrayado. Representan valores que el programa **NO DEBERÍA MODIFICAR**.

6. Para el caso de las **variables**, sus nombres se escriben en minúscula y separando las palabras con el caracter de subrayado. Representan valores que **PUEDEN** cambiar o sobreescribirse en un programa.

En la siguiente tabla se presentan nombres de variables y constantes adecuados y no adecuados para programas en Python:

#### Variables

|Ejemplo         |Correcto/Incorrecto                                       |
|:--------------:|:--------------------------------------------------------:|
|resultado       |Correcto                                                  |
| sumaAcumulada  |Incorrecto: las palabras se deben separar por (**`_`**)   |
| suma_parcial   |Correcto                                                  |
| x1             |Incorrecto: nombre no representativo                      |
| cosa2          |Incorrecto: nombre no representativo                      |

#### Constantes

|Ejemplo         |Correcto/Incorrecto                  |
|:--------------:|:-----------------------------------:|
| PI             |Correcto                             |
| NUMERO_E       |Correcto                             |
| gravedad       |Incorrecto: no usa mayúsculas        |
| numeroPi       |Incorrecto: no usa mayúsculas        |

Para comprender, de mejor manera, se presenta el siguiente ejemplo donde se calcula el perímetro de una circunferencia:

#### Ejemplo Variables y Constantes

A continuación se muestra un ejemplo donde se pretende obtener el perímetro de una circunferencia de radio 3

In [59]:
# Ejemplo 59: Se definen las constantes y variables a utilizar
NUMERO_PI = 3.14
radio = 3

# Se realiza los cálculos respectivos
perimetro_circunferencia = 2 * NUMERO_PI * radio

# Se entrega el resultado
perimetro_circunferencia

18.84

Del mismo modo, se presenta un nuevo ejemplo donde se desea calcular el perímetro de un rectángulo

In [60]:
# Ejemplo 60: Se definen las variables a utilizar
lado_a = 3
lado_b = 4

# Se realizan los cálculos respectivos
perimetro_rectangulo = 2*lado_a + 2*lado_b

# Se entrega el resultado
perimetro_rectangulo

14

---

## Actividad de cierre: reflexión

Antes de pasar al siguiente apunte, tómate unos minutos para responder estas tres preguntas. No se evalúan y no hay respuestas correctas: sirven para que te des cuenta de qué te quedó firme y qué conviene repasar. Responde en tu cuaderno o donde indique el equipo docente.

**1. ¿Qué aprendiste en este apunte?**

Resúmelo con tus palabras, sin volver a mirar el texto. Si puedes explicarlo sin leerlo, lo aprendiste.

**2. ¿Qué te resultó más difícil o desafiante?**

Nómbralo lo más concreto que puedas: un concepto, un ejemplo, un mensaje de error que no entendiste.

**3. ¿Cómo puedes aplicar esto en la vida real?**

Piensa en un problema concreto —de otro de tus ramos o de tu vida cotidiana— donde lo que viste aquí sea útil.

> **Pista para este apunte:** piensa en cualquier cálculo donde los decimales importen: tu promedio ponderado de notas, el resultado de un laboratorio de Física, el total de una boleta. Si los flotantes no son exactos, ¿qué consecuencias puede tener eso?


# Bibliografía

## Sobre la notación infija

Colaboradores de Wikipedia. (2019, 25 diciembre). *Notación de infijo*. Wikipedia, la enciclopedia libre. Recuperado 1 de agosto de 2022, de https://es.wikipedia.org/wiki/Notaci%C3%B3n_de_infijo

## Sobre notación de punto flotante

Borgwardt, M. (s. f.). *Números de punto flotante*. La Guía del Punto Flotante. Recuperado 1 de agosto de 2022, de http://puntoflotante.org/formats/fp/

Colaboradores de Wikipedia. (2022, 10 mayo). *Coma flotante*. Wikipedia, la enciclopedia libre. Recuperado 1 de agosto de 2022, de https://es.wikipedia.org/wiki/Coma_flotante

## Sobre tipos numéricos y operadores aritméticos

Ceder, V. L., Harms, D. K., & McDonald, K. M. (2010). The Quick Python Book. En *Built-in data types* (2.a ed., p. 19). Manning.

GeeksforGeeks. (2021, 1 octubre). *Python Data Types*. Recuperado 2 de agosto de 2022, de https://www.geeksforgeeks.org/python-data-types/

GeeksforGeeks. (2020, 30 agosto). *Python Arithmetic Operators*. Recuperado 2 de agosto de 2022, de https://www.geeksforgeeks.org/python-arithmetic-operators/

Python Software Foundation. (2022, 1 agosto). An Informal Introduction to Python. Python 3.10.5 documentation. Recuperado 1 de agosto de 2022, de https://docs.python.org/3/tutorial/introduction.html

## Estándar de formato
El Libro De Python. (2022). *Python PEP8*. Recuperado 1 de agosto de 2022, de https://ellibrodepython.com/python-pep8#convenciones-al-nombrar-elementos-camelcase-y-snake_case

Van Rossum, G., Warsaw, B., & Coghlan, N. (2001, 21 julio). *PEP 8 – Style Guide for Python Code*. Python Enhancement Proposals. Recuperado 1 de agosto de 2022, de https://peps.python.org/pep-0008/
